snap/local/files/run_nv_hostengine.sh

```diff
@@ -11,4 +11,11 @@ if [ -n "$nv_hostengine_port" ]; then
     args+=("-p" "$nv_hostengine_port")
 fi
 
+# Check if dcgm-exporter service is running and restart it
+# systemctl will not work in strict confinement
+dcgm_exporter_status=$(snapctl services dcgm.dcgm-exporter | awk '/dcgm\.dcgm-exporter/ {print $3}')
+if [ "$dcgm_exporter_status" == "active" ]; then
+    snapctl restart dcgm.dcgm-exporter
+fi
```

Comment:
Instead of doing this here I think we should try to create some [configure-hooks](https://snapcraft.io/docs/supported-snap-hooks#heading--the-configure-hook) where it restart dcgm-exporter and nv-hostengine when ports/address are changed (maybe into a different PR)

Reply:
I think you're right, lets do it in a different PR, maybe along with this enhancement #25

Reply:
Should we drop this restart in this PR then?

Reply:
As discussed in the santup, let's try to wrap things, so I'm ok with this approach and maybe do it later to automatically restart the services. My implementation of the charm is restarting the services anyway.